traffic_ops/testing/api/v4/coordinates_test.go

```diff
@@ -198,3 +238,174 @@ func DeleteTestCoordinates(t *testing.T) {
+	}
+
+	if len(coordinates) > 0 {
+		qparams = url.Values{}
+		qparams.Set("orderby", "id")
+		qparams.Set("limit", "1")
+		coordinatesWithLimit, _, err := TOSession.GetCoordinates(qparams, nil)
+		if err == nil {
+			if !reflect.DeepEqual(coordinates[:1], coordinatesWithLimit) {
+				t.Error("expected GET Coordinates with limit = 1 to return first result")
+			}
+		} else {
+			t.Error("Error in getting coordinates by limit")
+		}
+
+		if len(coordinates) > 1 {
+			qparams = url.Values{}
+			qparams.Set("orderby", "id")
+			qparams.Set("limit", "1")
+			qparams.Set("offset", "1")
+			coordinatesWithOffset, _, err := TOSession.GetCoordinates(qparams, nil)
+			if err == nil {
+				if !reflect.DeepEqual(coordinates[1:2], coordinatesWithOffset) {
+					t.Error("expected GET Coordinates with limit = 1, offset = 1 to return second result")
+				}
+			} else {
+				t.Error("Error in getting coordinates by limit and offset")
+			}
+
+			qparams = url.Values{}
+			qparams.Set("orderby", "id")
+			qparams.Set("limit", "1")
+			qparams.Set("page", "2")
+			coordinatesWithPage, _, err := TOSession.GetCoordinates(qparams, nil)
+			if err == nil {
+				if !reflect.DeepEqual(coordinates[1:2], coordinatesWithPage) {
+					t.Error("expected GET Coordinates with limit = 1, page = 2 to return second result")
+				}
+			} else {
+				t.Error("Error in getting coordinates by limit and page")
+			}
+		} else {
+			t.Errorf("only one Coordinates found, so offset functionality can't test")
+		}
+	} else {
+		t.Errorf("No Coordinates found to check pagination")
+	}
+
+	qparams = url.Values{}
+	qparams.Set("limit", "-2")
+	_, _, err = TOSession.GetCoordinates(qparams, nil)
+	if err == nil {
+		t.Error("expected GET Coordinates to return an error when limit is not bigger than -1")
+	} else if !strings.Contains(err.Error(), "must be bigger than -1") {
+		t.Errorf("expected GET Coordinates to return an error for limit is not bigger than -1, actual error: " + err.Error())
+	}
+
+	qparams = url.Values{}
+	qparams.Set("limit", "1")
+	qparams.Set("offset", "0")
+	_, _, err = TOSession.GetCoordinates(qparams, nil)
+	if err == nil {
+		t.Error("expected GET Coordinates to return an error when offset is not a positive integer")
+	} else if !strings.Contains(err.Error(), "must be a positive integer") {
+		t.Errorf("expected GET Coordinates to return an error for offset is not a positive integer, actual error: " + err.Error())
+	}
+
+	qparams = url.Values{}
+	qparams.Set("limit", "1")
+	qparams.Set("page", "0")
+	_, _, err = TOSession.GetCoordinates(qparams, nil)
+	if err == nil {
+		t.Error("expected GET Coordinates to return an error when page is not a positive integer")
+	} else if !strings.Contains(err.Error(), "must be a positive integer") {
+		t.Errorf("expected GET Coordinates to return an error for page is not a positive integer, actual error: " + err.Error())
+	}
+}
+
+func CreateTestCoordinatesWithInvalidName(t *testing.T) {
+	firstCoordinates := testData.Coordinates[0]
```

Comment:
This will segfault if there are no Coordinates in the testing data.

It's not so much a concern that someone will try to run the tests with modified data that's invalid, but the data is mutable after load, so other tests or even client methods that take slice references can put the data into a bad state. Which is actually something I ran into with the DSR tests where the client method was taking in references to DSRs and populating them with transient IDs for objects that would "stick" after the tests end.

Reply:
Fixed.